### ISSUE TITLE
Error in android support appcompat library project 

### ISSUE THREAD

**Issue by VladimirAdv** [dandar3/android-support-compat#1]
aidl.cpp:580] refusing to generate code from aidl file defining parcelable

Read more:
https://stackoverflow.com/questions/45675875/error-opening-android-support-appcompat-in-eclipse-from-dandar3

**Comment by dandar3** [comment]
Just to confirm, this is an issue that affects a number of Android Support Libraries version `26.0.0`:
* `android-support-compat`
* `android-support-media-compat`
* `android-support-v4`

And it is thrown by [`aidl.exe`](https://android.googlesource.com/platform/system/tools/aidl/+/master/aidl.cpp#580).

The classes declared in the `.aidl` files seem to exist in the JAR libraries, so maybe there is an issue with the Google ADT plugin not calling `aidl.exe` with the appropriate classpath to look for the `parcealable` files declared in the `.aidl`.

For the moment I would suggest removing them from those projects so you can compile and continue working, and I'll continue having a look at this. Not even sure at this stage whether there's any actual use for these `.aidl` the way they only declare a parcelable class that is already accessible from the JARs in CLASSPATH.

**Comment by dandar3** [comment]
The problem is that `aidl` tool logs an error to the console (line 580) that wasn't meant to be a terminal error, but ADT was designed to take external tool error statements as compiler errors. 

You can see them if you enable `Verbose` mode for `Preferences` > `Android` > `Build` > `Build output`.
```
[2017-08-17 22:02:10 - android-support-v4] AIDL: android/support/v4/media/session/MediaSessionCompat.aidl
[2017-08-17 22:02:10 - android-support-v4] U:\Android\android-sdk\build-tools\26.0.1\aidl.exe "-pU:\Android\android-sdk\platforms\android-26\framework.aidl" "-IYOUR_WORKSPACE\android-support-v4\src" "-IYOUR_WORKSPACE\android-support-v4\gen" "-IYOUR_WORKSPACE\android-support-media-compat\bin\aidl" "YOUR_WORKSPACE\android-support-v4\src\android\support\v4\media\session\MediaSessionCompat.aidl" "YOUR_WORKSPACE\android-support-v4\gen\android\support\v4\media\session\MediaSessionCompat.java" 
[2017-08-17 22:02:10 - android-support-v4] aidl.exe E 08-17 22:02:10  9120  9696 aidl.cpp:580] refusing to generate code from aidl file defining parcelable
```

`aidl` tool appears to only support interfaces ([aidl.cpp](https://android.googlesource.com/platform/system/tools/aidl/+/master/aidl.cpp#580)):

```
547 AidlError load_and_validate_aidl(
...
575  AidlDocument* parsed_doc = p.GetDocument();
576
577  unique_ptr<AidlInterface> interface(parsed_doc->ReleaseInterface());
578
579  if (!interface) {
580    LOG(ERROR) << "refusing to generate code from aidl file defining "
581                  "parcelable";
582    return AidlError::FOUND_PARCELABLE;
583  }
```
and it even has a startup parameter for `aidl.exe` (-b) to fail the process when parsing parcealable: 
```
-b         fail when trying to compile a parcelable.
```
that is turned off by default, but that is useless when called by ADT as it doesn't make any use of the result code (it should have, really).
```
717  if (aidl_err == AidlError::FOUND_PARCELABLE && !options.fail_on_parcelable_) {
718    // We aborted code generation because this file contains parcelables.
719    // However, we were not told to complain if we find parcelables.
720    // Just generate a dep file and exit quietly.  The dep file is for a legacy
721    // use case by the SDK.
722    write_java_dep_file(options, imports, io_delegate, "");
723    return 0;
724  }
```

**Comment by dandar3** [comment]
The 3 libraries mentioned earlier were released again (version 26.0.0).

PS: If you're not using this method already, you can easily import (or re-import) an entire library set using Team Project Sets, see the README.md notes for each library project and the video as the end of https://github.com/dandar3/android-support-library-README

**Comment by VladimirAdv** [comment]
Thanks for answer. In other project relating GPS i got this error. But i solve it, by deleting .aidl file and adding jar file in lib in linking project. Because only deleting generate another error...
